### PR TITLE
GFORMS-2383: Improve error message displayed while clicking upload fi…

### PR DESCRIPTION
**Pull request by lxol**
…le button without selecting file

### CODE REVIEW

File: app/uk/gov/hmrc/gform/upscan/UpscanController.scala [hunk 1]
```diff
@@ -196,11 +189,19 @@ class UpscanController(
 
         val flash = errorCode match {
           case Some("InvalidArgument") =>
+            val formModel = formModelOptics.formModelVisibilityOptics.formModel
+            val formComponent = formModel.fcLookup.get(formComponentId)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/upscan/UpscanController.scala:193]: This is done on 200-201, it deserves some refactoring

**Review by josef-vlach — CHANGES_REQUESTED** [review]
There is no consistency for js and non-js users when file is not selected.

js users see error: Select a file
non-js users see error: Select a file. It must be a PDF, JPG, XLSX, ODS, DOCX, ODT, PPTX, ODP.

Also for non-js users upload of empty file case is not properly detected anymore.

**Comment by lxol** [comment]
nonjs and js can't be consistently. We have different ACs for nonjs

**Comment by lxol** [comment]
Empty file detection should stay the same. I will have a look tomorrow

**Comment by lxol** [comment]
>Also for non-js users upload of empty file case is not properly detected anymore.
I think it has the same behavior for empty file. Looks like locally upscan returns a generic error that I changes according to ACs. I believe it will be the same in QA.

**Review by josef-vlach — APPROVED** [review]
Reusing is `error_action_redirect` is neat solution 👍

**Pull request merged**